src/clojure/nrepl/cmdline.clj

```diff
@@ -239,59 +244,193 @@
              options
              options))
 
+(defn args->cli-options
+  "Takes CLI args list and returns vector of parsed options map and
+  remaining args."
+  [args]
+  (let [[options _args] (split-args (expand-shorthands args))
+        merge-config (partial merge config/config)
+        options (-> options
+                    (keywordize-options)
+                    (parse-cli-values)
+                    (merge-config))]
+    [options _args]))
+
+(defn help-command
+  "Displays help copy to the user and exits the program"
+  []
+  (display-help)
+  (exit 0))
+
+(defn version-command
+  "Displays the version to the user and exits the program"
+  []
+  (println (:version-string version/version))
+  (exit 0))
+
+(defn get-transport-option
```

Comment:
Does anything related to the config need to public?

Reply:
For my upcoming suprepl project I had to duplicate a lot of the configuration resolution. Making `get-conn-options` and `get-server-options` public would solve that. 

Functions such as `get-transport-option`, `get-handler-option`, `get-ack-port`, and `get-greeting-option` can be shortened in name and made private.

Reply:
That's fine by me. Generally the smaller the API is, the easier it's going to be to keep it stable.